### PR TITLE
Add Support for insecure updates over HTTP

### PR DESCRIPTION
**Pull request by mdhorn**
Fixes Issue: #478

Changes proposed in this pull request:
- Support insecure updates over HTTP
- swupd: identify certificate errors
- Ensure we default to software auto update for UI

### CODE REVIEW

File: clr-installer/main.go [hunk 1]
```diff
@@ -270,6 +266,14 @@ func main() {
 		md.SwupdMirror = options.SwupdMirror
 	}
 
+	if options.AllowInsecureHTTPSet {
+		md.AllowInsecureHTTP = options.AllowInsecureHTTP
+	}
+
+	if options.SwupdContentURL != "" && swupd.IsValidMirror(options.SwupdContentURL, md.AllowInsecureHTTP) == false {
+		fatal(errors.Errorf("swupd-contenturl %s must use HTTPS or FILE protocol", options.SwupdContentURL))
```
Comment (jwakre) [inline clr-installer/main.go]: With rebase this becomes return ....

File: swupd/swupd.go [hunk 2]
```diff
@@ -539,6 +539,9 @@ func setMirror(swupdArgs []string, t string) (string, error) {
 // SetHostMirror executes the "swupd mirror" to set the Host's mirror
 func SetHostMirror(url string, allowInsecureHTTP bool) (string, error) {
 	if urlErr := network.CheckURL(url); urlErr != nil {
+		if strings.Contains(urlErr.Error(), "60") {
+			return "", fmt.Errorf(utils.Locale.Get("SSL certificate problem"))
```
Comment (jwakre) [inline swupd/swupd.go]: Use `errors.Errorf` to enable stack trace.

**Comment by jwakre** [comment]
> When I try using an HTTP URL in the GUI, I get an error. Steps to reproduce
> 
>     1. Launch GUI
> 
>     2. Go to Software Updater Configuration
> 
>     3. Check `allow installation over insecure...`
> 
>     4. Enter the following URL: "http://cdn.download.clearliunx.org/update/"
> 
>     5. Hit Confirm
> 
> 
> I get the following error: Server does not report any verison

This URL seems to be problematic for curl. I'll try another one.

File: swupd/swupd.go (hunk 2 above)
Comment (mdhorn) [reply swupd/swupd.go]: In this instance, we do not want the stack trace as this string is display to the user in the UI.

**Pull request merged**